lib/src/config_provider/config_spec.dart

```diff
@@ -0,0 +1,811 @@
+import 'package:logging/logging.dart';
+import 'package:yaml/yaml.dart';
+
+final _logger = Logger('ffigen.config_provider.config');
+
+/// A container object for a ConfigSpec Object.
+class ConfigSpecNode<TE> {
```

Comment:
Should this be:

```dart
/// An individual value in a config for a specific [path] instantiated from a [ConfigSpec].
///
/// A config value contains both the [value] that users of the configuration would want
/// to use, as well as the [rawValue] that was provided as input to the configuration.
class ConfigValue<TE> {
```

(e.g. what is it a container for? I guess the rawValue. However, it also contains the path.)

Reply:
Well, I used the Tree -> TreeNode analogy, but I guess this seems better.